pudb/run.py

```diff
@@ -1,9 +1,15 @@
-COMMAND = {"zsh": "_command_names -e"}
+COMMAND = {"zsh": "_pre_run"}
 PREAMBLE = {
     "zsh": """\
 _script_args() {
   _arguments -S -s '(-)1:script_args:_files -g "*.py"' '*: :_files'
 }
+
+_pre_run() {
+  # https://github.com/zsh-users/zsh/blob/master/Completion/Unix/Command/_sudo#L47-L51
+  _comp_priv_prefix=($words[1] -n ${(kv)opt_args[(I)(-[ugHEP]|--(user|group|set-home|preserve-env|preserve-groups))]})
```

Comment:
I'm a little lost here. What does this mean/do?

Reply:
See the commit message of <https://github.com/zsh-users/zsh/commit/359e0dae2bded75eebf7fe3b3cf919e1dd44c333#diff-b3f104490c12fc9f1ef7365c016b1b51c1cc409454417bec00f98e96f6d1f44a>:

```
…sudo

* Update _command_names to add **/sbin to PATH when not otherwise overridden
  and _comp_priv_prefix is set. This ensures that, on systems where
  unprivileged users don't have **/sbin in their PATHs, we can still complete
  those commands when they try to run them with e.g. sudo

* Update functions for privilege-gaining commands to use the aforementioned
  mechanism

* Fix some other minor issues in _pfexec and _doas

Changes from the posted diff:

* Fix a minor redundancy in _doas

* Add _c_p_p=( '' ) hack to _su
```